Source/BestProjectEver/Private/Character/BPE_PlayerCharacter.cpp

```diff
@@ -236,45 +268,126 @@ bool ABPE_PlayerCharacter::Server_EquipWeapon_Validate(ABPE_Weapon* WeaponToEqui
 }
 
 //----------------------------------------------------------------------------------------------------------------------
-void ABPE_PlayerCharacter::SetEquippedWeapon(ABPE_Weapon* WeaponToEquip, ABPE_Weapon* LastWeapon)
+void ABPE_PlayerCharacter::HandleEquipWeapon(ABPE_Weapon* WeaponToEquip)
 {
-	if(IsValid(LastWeapon))
+	bool bIsColorInInventory = IsValid(FindWeaponWithColorType(WeaponToEquip->GetColorType()));
+	if(bIsColorInInventory)
 	{
-		LastWeapon->SetState(EWeaponState::Idle);
-		
-		const FDetachmentTransformRules DetachRules(EDetachmentRule::KeepWorld, true);
-		LastWeapon->DetachFromActor(DetachRules);
-		LastWeapon->SetOwner(nullptr);
+		/*Check the ammo to reload current weapon or add to inventory this new weapon if it is more powerful*/
+	}
+	else
+	{
+		PickupWeapon(WeaponToEquip);
+		Server_SetAsCurrentWeapon(WeaponToEquip);
```

Comment:
Recuerda que solo llamamos RPC de Server desde cliente. Aqui pareciera que HandleEquipWeapon se llama desde server, verdad? Pensando en que PickupWeapon solo ejecuta logica en HasAuthority

Reply:
![image](https://user-images.githubusercontent.com/85327912/172199091-fef3a52f-9b49-4477-9ab2-775349f4a3f3.png)

Reply:
![image](https://user-images.githubusercontent.com/85327912/172199179-960b08da-310f-4e81-8925-2b415a036018.png)
Listo lo cambie pero igual deje el RPC para llamarlo desde equip next and previous weapon para que desde el servidor se actualice que arma estoy cargando

Reply:
![image](https://user-images.githubusercontent.com/85327912/172199385-5367e7b3-55e5-4540-93a6-bf12480bde5c.png)

aqui